checks/pinned_dependencies.go

```diff
@@ -161,6 +167,44 @@ func addPinnedResult(r *pinnedResult, to bool) {
 	}
 }
 
+func dataAsWorkflowResultPointer(data FileCbData) *worklowPinningResult {
+	pdata, ok := data.(*worklowPinningResult)
+	if !ok {
+		// panic if it is not correct type
+		panic("type need to be of worklowPinningResult")
```

Comment:
Panic is recommended, please change it to an error

Reply:
> Panic is recommended, please change it to an error

@naveensrinivasan Sorry I'm bit lost in the comment is `panic NOT recomended` or `panic IS recommended` ?

Reply:
The reason for using `panic` is because existing code is also using `panic` for this kind of data checking https://github.com/ossf/scorecard/blob/d007f73e07edb900bd4e8146a035094dfbb590ee/checks/pinned_dependencies.go#L212

Reply:
I apologize for stating `panic is recommended`, `panic` is not recommended. https://golang.org/doc/effective_go#panic 

If the old code had that then we would have to fix that.

Reply:
the `panic` will never be triggered unless someone is debugging their code. The `panic` is only used for troubleshooting purposes: it's triggered if the developer uses the wrong structure in their code. 

I think it's fine to leave a `panic()` here as we want to catch the problem early during development, and we don't expect the program to `recover`.

Reply:
Sounds good. Thanks!